### PR TITLE
[ISSUE #544] Fix bug, concurrent map read and map write local offest file

### PR DESCRIPTION
**Pull request by feiquan123**
## What is the purpose of the change

 avoid `fatal error: concurrent map read and map write`

## Brief changelog

use sync.Map to save` localFileOffsetStore.OffsetTable` info.

## Verifying this change

`go test . -v --run TestNewLocalFileOffsetStore` pass
`go test . -v --run TestLocalFileOffsetStore` pass

### CODE REVIEW

**Comment by feiquan123** [comment]
I had verified this feature. It working in produce env. 

current `gomod` file:
```go
replace (
	github.com/apache/rocketmq-client-go/v2 => github.com/feiquan123/rocketmq-client-go/v2 v2.2.0
)
```

__When will next version be released? I want to use__ `github.com/apache/rocketmq-client-go/v2`

**Comment by xujianhai666** [comment]
why ther are some change to readme.md ? plz keep code clean. @feiquan123

File: README.md [hunk 1]
```diff
@@ -53,3 +53,4 @@ For 2.0.0 version, it supports:
 ----------
 ## License
   [Apache License, Version 2.0](http://www.apache.org/licenses/LICENSE-2.0.html) Copyright (C) Apache Software Foundation
+
```
Comment (xujianhai666) [inline README.md:56]: extra line, remove that

Comment (feiquan123) [reply README.md]: sorry, the space line autoadd by vscode.😞

**Comment by xujianhai666** [comment]
lastly, rename pull request title reference by  https://github.com/apache/rocketmq/wiki/RIP-14-RocketMQ-Community-Operation-Conventions

**Comment by xujianhai666** [comment]
remember to squish commits to one

**Review by ShannonDing — APPROVED** [review]

**Pull request merged**